### PR TITLE
camel case style properties

### PR DESCRIPTION
**Pull request by substack**
This patch fixes the display of camel-cased style properties showing up in the html (https://github.com/Matt-Esch/virtual-dom/issues/359).

I had to update some tests to check for properties using the dashed property names instead of camel-cased properties, but the changes are relatively minor.

### CODE REVIEW

**Comment by Matt-Esch** [comment]
I'm not sure I understand why virtual-dom would need to change the casing of the properties. I understand that the html rendering is producing the wrong thing, but this isn't the job of virtual-dom to accommodate. virtual-dom is based purely on properties of the dom nodes in a browser agnostic fashion. min-document should be patched to generate correct html from the virtual dom node.

File: test/style-camel-case.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+var test = require("tape")
+var h = require("../h.js")
+var diff = require("../diff.js")
+var patch = require("../patch.js")
+var createElement = require("../create-element.js")
+
+test("convert camel-case properties to dashes in string form", function (assert) {
```
Comment (Raynos) [inline test/style-camel-case.js]: This is a good test :)

We should fix `.outerHTML` and `.toString()` in `min-document`.

File: test/nested-properties.js [hunk 2]
```diff
@@ -25,18 +25,18 @@ test("dom node style", function (assert) {
...
     var rootNode = render(a)
     assert.equal(rootNode.style.border, style("border", "none"))
-    assert.equal(rootNode.style.className, style("className", "oops"))
+    assert.equal(rootNode.style["class-name"], style("className", "oops"))
     assert.equal(rootNode.style.display, style("display", "none"))
     var s1 = rootNode.style
     var equalNode = render(b)
     assert.equal(equalNode.style.border, style("border", "1px solid #000"))
-    assert.equal(equalNode.style.className, style("className", "oops"))
+    assert.equal(equalNode.style["class-name"], style("className", "oops"))
```
Comment (Raynos) [inline test/nested-properties.js]: I think this change is "bad". I don't think this change should happen at this level of the abstraction.

**Comment by Raynos** [comment]
Agreed.

Doing `elem.style.className = x;` and `elem.style['class-name'] = x;` are perfectly valid DOM expression.

It's `min-document` that does the wrong thing and we should fix that instead.

**Comment by substack** [comment]
closing since this issue is handled upstream in min-document

**Pull request closed without merging**